python/cudf/cudf/core/reshape.py

```diff
@@ -73,7 +103,7 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
                 objs[idx] = o.to_frame(name=name)
 
         for idx, o in enumerate(objs):
-            if idx == 0:
+            if idx == 0 and not ignore_index:
```

Comment:
Python will skip evaluating the second clause if the first clause is first, so micro-optimization:
```suggestion
            if not ignore_index and idx == 0:
```